clang/test/SemaSYCL/implicit_kernel_type.cpp

```diff
@@ -1,47 +1,16 @@
-// RUN: %clang_cc1 -fsycl -fsycl-is-device -fsycl-int-header=%t.h -fsyntax-only -verify %s -Werror=sycl-strict -DERROR
-// RUN: %clang_cc1 -fsycl -fsycl-is-device -fsycl-int-header=%t.h -fsyntax-only -verify %s  -Wsycl-strict -DWARN
-// RUN: %clang_cc1 -fsycl -fsycl-is-device -fsycl-int-header=%t.h -fsycl-unnamed-lambda -fsyntax-only -verify %s  -Werror=sycl-strict
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -internal-isystem %S/Inputs -fsycl-int-header=%t.h -fsyntax-only -verify %s -Werror=sycl-strict -DERROR
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -internal-isystem %S/Inputs -fsycl-int-header=%t.h -fsyntax-only -verify %s  -Wsycl-strict -DWARN
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -internal-isystem %S/Inputs -fsycl-int-header=%t.h -fsycl-unnamed-lambda -fsyntax-only -verify %s  -Werror=sycl-strict
 
-// SYCL 1.2 Definitions
-template <typename name, typename Func>
-__attribute__((sycl_kernel)) void sycl_121_single_task(Func kernelFunc) {
-  kernelFunc();
-}
-
-class event {};
-class queue {
-public:
-  template <typename T>
-  event submit(T cgf) { return event{}; }
-};
-class auto_name {};
-template <typename Name, typename Type>
-struct get_kernel_name_t {
-  using name = Name;
-};
-class handler {
-public:
-  template <typename KernelName = auto_name, typename KernelType>
-  void single_task(KernelType kernelFunc) {
-    using NameT = typename get_kernel_name_t<KernelName, KernelType>::name;
-#ifdef __SYCL_DEVICE_ONLY__
-    sycl_121_single_task<NameT>(kernelFunc);
-#else
-    kernelFunc();
-#endif
-  }
-};
-// -- /Definitions
-
-#ifdef __SYCL_UNNAMED_LAMBDA__
-// expected-no-diagnostics
-#endif
```

Comment:
This condition was no longer true. With the updated test, passing `-fsycl-unnamed-lambda` caused the following error: Passing of kernel functions by reference is a SYCL 2020 extension. I added an #if check to capture that.